src/proto/codec.rs

```diff
@@ -142,16 +140,15 @@ impl Decoder for WebSocketProtocol {
                 ensure_buffer_has_space!(src, offset + 8);
                 // SAFETY: The ensure_buffer_has_space call has validated this
                 // A conversion from 8 u8s to a u64 cannot fail
-                payload_length = u64::from_be_bytes(unsafe {
-                    src.get_unchecked(2..10).try_into().unwrap_unchecked()
-                }) as usize;
+                payload_length =
+                    u64::from_be_bytes(unsafe { src.get_unchecked(2..10).try_into().unwrap() })
```

Comment:
It's obvious to the optimizer that the length is 8

Reply:
Should be correct, same as the other case.